tenets/codelingo/go/unsafe-goroutine/codelingo.yaml

```diff
@@ -0,0 +1,33 @@
+tenets:
+  - name: find-funcs
+    doc: Example tenet that finds unsafe variables in goroutines.
+    flows:
+      codelingo/review:
+        comment: This variable is not safe to use in this goroutine.
+    query: |
+      import codelingo/ast/go
+
+      go.func_decl(depth = any):
+      # Repeated code under go.for_stmt and go.range_stmt - waiting for macrofacts
+        any_of:
+          go.range_stmt(depth = any):
+            go.go_stmt(depth = any):
+              go.expr_stmt(depth = any):
+                @review.comment
+                go.ident:
+                  name as identName
+              exclude:
+                go.names(depth = any):
```

Comment:
I'm pretty sure this one shouldn't be at `(depth = any)`, or else you'll get a false negative for functions this, because the `unsafeItem` _does_ exist in a `go.names`. 
```
func uncaughtUnsafe() {
    for _, unsafeItem := range list {
        go func() {
            go func(unsafeItem int) {}(unsafeItem)
            // do something with item 
            unsafeItem
        }()
     }
}
```

Reply:
Also, what about variables that are declared in the go routine? Won't they give you a false positive?

```
func falsePositive() {
    for _, _ := range list {
        go func() {
            safeItem := 1
            // do something with item 
            safeItem
        }()
     }
}
```

Reply:
The go.names(depth = any) was supposed to be removed, as it was from the block below. Fixing that and working on a way to prevent that false positive case

Reply:
You should be able to resolve the false positive with another `exclude`.